File: android/src/main/java/com/whiteguru/capacitor/plugin/videoeditor/TranscodeSettings.java
package com.whiteguru.capacitor.plugin.videoeditor;

public class TranscodeSettings {

    private int height = 0;
    private int width = 0;
    private boolean keepAspectRatio = true;
    private int fps = 30;

    public TranscodeSettings() {}

    public TranscodeSettings(int height, int width, boolean keepAspectRatio, int fps) {
        if (height < 0) {
            throw new IllegalArgumentException("Parameter height cannot be negative");
        }

        if (width < 0) {
            throw new IllegalArgumentException("Parameter width cannot be negative");
        }

        if (fps < 1) {
            throw new IllegalArgumentException("Parameter fps cannot be lower than 1");
        }

        this.height = height;
        this.width = width;
        this.keepAspectRatio = keepAspectRatio;
        this.fps = fps;
    }

    public int getHeight() {
        return height;
    }

    public void setHeight(int height) {
        if (height < 0) {
            throw new IllegalArgumentException("Parameter height cannot be negative");
        }

        this.height = height;
    }

    public int getWidth() {
        return width;
    }

    public void setWidth(int width) {
        if (width < 0) {
            throw new IllegalArgumentException("Parameter width cannot be negative");
        }

        this.width = width;
    }

    public boolean isKeepAspectRatio() {
        return keepAspectRatio;
    }

    public void setKeepAspectRatio(boolean keepAspectRatio) {
        this.keepAspectRatio = keepAspectRatio;
    }

    public int getFps() {
        return fps;
    }

    public void setFps(int fps) {
        if (fps < 1) {
            throw new IllegalArgumentException("Parameter fps cannot be lower than 1");
        }

        this.fps = fps;
    }
}

File: ios/Plugin/TranscodeSettings.swift
import Foundation;

enum TranscodeSettingsError: Error {
    case invalidArgument(message: String)
}

public class TranscodeSettings: NSObject
{
    private var height: Int = 0;
    private var width: Int = 0;
    private var keepAspectRatio: Bool = true;
    private var fps: Int = 30;

    init(height: Int, width: Int, keepAspectRatio: Bool, fps: Int) throws
    {
        if (height < 0)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter height cannot be negative");
        }
        if (width < 0)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter width cannot be negative");
        }
        if (fps < 1)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter fps cannot be lower than 1");
        }

        self.height = height;
        self.width = width;
        self.keepAspectRatio = keepAspectRatio;
        self.fps = fps;
    }
    
    func getHeight()->Int
    {
        return self.height;
    }
    
    func setHeight(_ height: Int) throws
    {
        if (height < 0)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter height cannot be negative");
        }
        self.height = height;
    }
    
    func getWidth()->Int
    {
        return self.width;
    }
    
    func setWidth(_ width: Int) throws
    {
        if (width < 0)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter width cannot be negative");
        }
        self.width = width;
    }
    
    func isKeepAspectRatio()->Bool
    {
        return self.keepAspectRatio;
    }
    
    func setKeepAspectRatio(_ keepAspectRatio: Bool)
    {
        self.keepAspectRatio = keepAspectRatio;
    }

    func getFps()->Int
    {
        return self.fps;
    }

    func setFps(_ fps: Int) throws
    {
        if (fps < 1)
        {
            throw TranscodeSettingsError.invalidArgument(message: "Parameter fps cannot be lower than 1");
        }
        self.fps = fps;
    }
}
